packages/components/src/components/entities/EntityInsertPanel.tsx

```diff
@@ -530,9 +533,13 @@ export class EntityInsertPanelImpl extends React.Component<Props, StateProps> {
                             return this.renderParentTypes(dataType);
                         })}
                         <div className="entity-insert--header">
-                            {parentDataTypes.map(dataType => {
+                            {!combineParentTypes && parentDataTypes.map(dataType => {
                                 return this.renderAddEntityButton(dataType);
                             })}
+                            {combineParentTypes &&
+                                // Just grabbing first parent type for the name
+                                this.renderAddEntityButton(parentDataTypes.get(0))
+                            }
```

Comment:
I think this is a good place for a ternary operator:
```suggestion
                            {combineParentTypes ?  
                              // Just grabbing first parent type for the name
                                this.renderAddEntityButton(parentDataTypes.get(0))
                                : parentDataTypes.map(dataType => {
                                return this.renderAddEntityButton(dataType);
                            })}             
```
I also am a little wary of the "just grab the first parent".  It relies on the user knowing that this will happen so at the very least deserves a comment on the properties.

Reply:
Incorporated this refactor and added comment by prop as suggested